pubsub/README.md

```diff
@@ -112,30 +115,34 @@ message Message {
 }
 ```
 
-The `from` field denotes the author of the message, note that this is not
-necessarily the peer who sent the RPC this message is contained in. This is
-done to allow content to be routed through a swarm of pubsubbing peers.
+The `optional` fields may be omitted, depending on the
+[signature policy](#message-signing) and
+[message ID function](#message-identification).
 
-The `data` field is an opaque blob of data, it can contain any data that the
-publisher wants it to.
+The `from` field denotes the author of the message. This is the peer who
+initially authored the message, and NOT the peer who propagated it. Thus, as
+the message is routed through a swarm of pubsubbing peers, the original
+authorship is preserved.
 
 The `seqno` field is a 64-bit big-endian uint that is a linearly increasing
 number that is unique among messages originating from each given peer. No two
 messages on a pubsub topic from the same peer should have the same `seqno`
-value, however messages from different peers may have the same sequence number,
-so this number alone cannot be used to address messages. Notably the
-'timecache' in use by the go implementation contains a `message_id`,
-which is constructed from the concatenation of the `seqno` and the `from`
-fields. This `message_id` is then unique among messages. It was also proposed
-in [#116](https://github.com/libp2p/specs/issues/116) to use a `message_hash`,
-however, it was noted: "a potential caveat with using hashes instead of seqnos:
-the peer won't be able to send identical messages (e.g. keepalives) within the
-timecache interval, as they will get rejected as duplicates."
+value, however messages from different peers may (and likely will) have the same
+sequence number. In other words, this number is not globally unique. It is used
```

Comment:
Better to say "It _may_ be used..."

Reply:
Also let's specify it is optional.

Reply:
Adjusted text. The spec already says it's optional, but it wouldn't hurt to put the optionality attribute in the prose for each field. Will do that.